beacon-chain/sync/validate_attester_slashing.go

```diff
@@ -50,7 +51,7 @@ func (s *Service) validateAttesterSlashing(ctx context.Context, pid peer.ID, msg
 	if err != nil {
 		return pubsub.ValidationIgnore, err
 	}
-	if err := blocks.VerifyAttesterSlashing(ctx, headState, slashing); err != nil {
+	if _, err := blocks.ProcessAttesterSlashing(ctx, headState, slashing, v.SlashValidator); err != nil {
```

Comment:
This mutates a beacon state and slashes a validator in vain, since the function `VerifyAttesterSlashing` anyway is called in a few other places (like RPC methods) and in particular is called within `ProcessAttesterSlashing` I believe that the right thing to do would be to perform the extra check within `VerifyAttesterSlashing` and if performance is an issue, this function can be refactored to return the validators that need to be slashed in the state (so that `ProcessAttesterSlashing` can use them)